docs/docs/adrs/adr-013-equivocation-slashing.md

```diff
@@ -0,0 +1,169 @@
+
+## Proposed solution
+As a first iteration, in order to implement the slashing functionality as soon as possible, we propose an aggressive (i.e., would slash more delegators than in the single-chain case) slashing  approach because this allows for a straight-forward implementation.
+In this approach, at the moment we receive evidence, at _evidence height_, we:
+1. slash all the undelegations and redelegations using `slashFactor`;
+2. slash all delegations using as voting `power` the sum of the voting power of the misbehaving validator and the power of all the ongoing undelegations and redelegations.
+
+**Evidence expiration:** Additionally, in this approach and because we cannot infer the actual time of the evidence (i.e., timestamp in evidence could contain anything), we do not consider _evidence expiration_ and hence old evidence is never ignored (e.g., we would act on 3 year-old evidence).
+
+### Implementation
+The aggressive approach allows for multiple simplifications. Specifically, we intend to introduce the following logic in the [HandleConsumerDoubleVoting](https://github.com/cosmos/interchain-security/pull/1232) method:
+```go
+undelegationsTotalAmount := sdk.NewInt(0)
+for _, v := range k.stakingKeeper.GetUnbondingDelegationsFromValidator(ctx, validatorAddress) {
+    for _, entry := range v.Entries {
+        undelegationsTotalAmount = undelegationsTotalAmount.Add(entry.InitialBalance)
+    }
+}
+
+redelegationsTotalAmount := sdk.NewInt(0)
+for _, v := range k.stakingKeeper.GetRedelegationsFromSrcValidator(ctx, validatorAddress) {
+    for _, entry := range v.Entries {
+        redelegationsTotalAmount = redelegationsTotalAmount.Add(entry.InitialBalance)
+    }
+}
+
+infractionHeight := 0
+totalPower := validators voting Power + undelegationsTotalAmount.Add(redelegationsTotalAmount)
+slashFraction := k.slashingKeeper.SlashFractionDoubleSign(ctx)
+
+k.stakingKeeper.Slash(ctx, validatorConsAddress, infractionHeight, totalPower, slashFraction, DoubleSign)
+```
+
+**Infraction height:** We provide a zero `infractionHeight` to the [Slash](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L33) method in order to slash all ongoing undelegations and redelegations (see checks in [Slash](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L92), [SlashUnbondingDelegation](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L195), and [SlashRedelegation](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L249)).
+
+**Power:** We pass the sum of the voting power of the misbehaving validator and the power of all the ongoing undelegations and redelegations. This is a slightly more aggressive approach than just providing the voting `power` at the evidence height. If we assume that the `slashFactor` is 5%, then the `power` we pass is `validatorPower + totalPower(undelegations) + totalPower(redelegations)`.
```

Comment:
> This is a slightly more aggressive approach than just providing the voting `power` at the evidence height.

What do you mean by "voting `power` at the evidence height"? Is the power when the evidence was received? Isn't that the same as "the voting power of the misbehaving validator" from above?

Reply:
Wanted to write "the voting `power` at the infraction height." instead. This is not necessarily the case so I removed the sentence altogether to avoid confusion.